### ISSUE TITLE
[BUG][typescript-angular] `anyOf` generates empty `interface` model

### ISSUE THREAD

**Issue by snebjorn** [OpenAPITools/openapi-generator#13428]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

`anyOf` schema definitions generates an empty interface model.
```jsonc
{
  "components": {
    "schemas": {
      "foo": {
        "type": "object",
        "properties": {
          "bar": {
            "anyOf": [{ "type": "number" }, { "type": "string" }] // <------- this
          }
        }
      }
    }
  }
}
```
Generates this model
```ts
export interface FooBar {  // <--- this doesn't represent type `number | string`
}
```
```ts
import { FooBar } from './fooBar';

export interface Foo { 
    bar?: FooBar;
}
```

##### openapi-generator version

This is a regression starting from v6.0.0, it worked in v5.4.0

[Output v6.1.0](https://github.com/snebjorn/openapi-gen-bug/tree/master/output-6.1.0/anyof-bug)
[Output v5.4.0](https://github.com/snebjorn/openapi-gen-bug/tree/master/output-5.4.0/anyof-bug)

##### OpenAPI declaration file content or url

https://github.com/snebjorn/openapi-gen-bug/blob/master/anyof-bug.json
```json
{
  "openapi": "3.0.0",
  "info": {
    "title": "",
    "version": ""
  },
  "paths": {
    "/Foos": {
      "get": {
        "responses": {
          "200": {
            "description": "",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/foo"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "foo": {
        "type": "object",
        "properties": {
          "bar": {
            "anyOf": [{ "type": "number" }, { "type": "string" }],
            "description": "This is the description for bar"
          }
        }
      }
    }
  }
}
```

##### Generation Details

Generated using only default values, see https://github.com/snebjorn/openapi-gen-bug/blob/59eb4ba55e91f8444add450016fbd5666c04e2ee/openapitools.json#L25-L29

##### Steps to reproduce

1. Clone https://github.com/snebjorn/openapi-gen-bug
2. Run yarn install
3. Run yarn gen
4. Observe empty interface in https://github.com/snebjorn/openapi-gen-bug/blob/master/output-6.1.0/anyof-bug/model/fooBar.ts

##### Related issues/PRs

Couldn't find any related to empty interfaces but a lot of issues popped up regarding `anyOf`, `allOf` and `oneOf`

##### Suggest a fix

Not exactly sure why `bar?: number | string | null;` was changed to `bar?: FooBar;`. Interfaces can't extend primitive types.
```ts
// An interface cannot extend a primitive type like 'string'; an interface can only extend named types and classes ts(2840)
interface FooBar extends string, number {}
```
It could be turned into a type
```ts
export type FooBar = string | number;
```

However there is also the issue of the `description`. The description is moved to the FooBar interface/type, but the description is describing the property `bar` which now looks like this
```ts
export interface Foo { 
    bar?: FooBar; // <--- note the missing description. The description is on `FooBar` which I believe is incorrect.
}
```
It should look like this
```ts
export interface Foo { 
    /**
     * This is the description for bar
     */
    bar?: FooBar;
}
```

**Comment by snebjorn** [comment]
@wing328 I've tracked down the issue to being related to the flattening of components which you implemented in #12175

As described in the issue. Flattening of types that use primitive types is troublesome as it cannot be turned into an `interface`. This is the case for TypeScript, Java and C# (I assume many others as well).

TL;DR;
``"anyOf": [{ "type": "number" }, { "type": "string" }],`` cannot be turned into `interface FooBar extends string, number`.

Why is the components flattened? What problem does it solve?

**Comment by PabloFNK** [comment]
Any fix for this issue? We are facing this exact problem.

**Comment by egorwow70** [comment]
@snebjorn 

Hello, any updates? Experiencing the same issue with 7.5.0

**Comment by snebjorn** [comment]
Sorry, I stopped working on this a while ago. Feel free to pick it up
